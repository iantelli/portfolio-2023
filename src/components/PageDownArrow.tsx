import { motion } from "framer-motion";
import Icon from "./Icon";
import { RiArrowDropDownLine } from "react-icons/ri";

export default function PageDownArrow() {
  return (
    <motion.div
      className="flex justify-center items-center w-full h-full mt-12 md:mt-24"
      initial={{ opacity: 0, y: 50 }}
      animate={{ opacity: 1, y: 0 }}
      transition={{ delay: 0.3, duration: 0.5 }}
    >
      <motion.div animate={{ y: [0, 10, 0] }} transition={{ repeat: Infinity, duration: 1.5 }}>
        <Icon
          aria={"link to project showcase"}
          link={"#projects"}
          icon={
            <RiArrowDropDownLine className="h-24 w-24 text-rosePineDawn-love dark:text-rosePine-foam" />
          }
        />
      </motion.div>
    </motion.div>
  );
}
