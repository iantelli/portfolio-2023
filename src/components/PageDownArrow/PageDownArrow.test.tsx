import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import PageDownArrow from ".";

describe("PageDownArrow", () => {
  it("renders PageDownArrow component", () => {
    const component = render(<PageDownArrow />);
    expect(component).toMatchSnapshot();
  });
});
