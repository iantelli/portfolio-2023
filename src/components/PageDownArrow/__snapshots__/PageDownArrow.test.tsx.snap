// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageDownArrow > renders PageDownArrow component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center items-center w-full h-full mt-12 md:mt-24"
        style="opacity: 0; transform: translateY(50px) translateZ(0);"
      >
        <div>
          <a
            aria-label="link to project showcase"
            href="#projects"
          >
            <div
              class="flex items-center justify-center h-auto w-auto"
            >
              <svg
                class="h-24 w-24 text-rosePineDawn-love dark:text-rosePine-foam"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 15l-4.243-4.243 1.415-1.414L12 12.172l2.828-2.829 1.415 1.414z"
                  />
                </g>
              </svg>
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center items-center w-full h-full mt-12 md:mt-24"
      style="opacity: 0; transform: translateY(50px) translateZ(0);"
    >
      <div>
        <a
          aria-label="link to project showcase"
          href="#projects"
        >
          <div
            class="flex items-center justify-center h-auto w-auto"
          >
            <svg
              class="h-24 w-24 text-rosePineDawn-love dark:text-rosePine-foam"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 15l-4.243-4.243 1.415-1.414L12 12.172l2.828-2.829 1.415 1.414z"
                />
              </g>
            </svg>
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
